src/d3d9/d3d9_adapter.cpp

```diff
@@ -768,30 +770,29 @@ namespace dxvk {
 
     // Walk over all modes that the display supports and
     // return those that match the requested format etc.
-    DEVMODEW devMode = { };
-    devMode.dmSize = sizeof(DEVMODEW);
+    wsi::WsiMode devMode = { };
 
     uint32_t modeIndex = 0;
 
     const auto forcedRatio = Ratio<DWORD>(options.forceAspectRatio);
 
-    while (GetMonitorDisplayMode(GetDefaultMonitor(), modeIndex++, &devMode)) {
+    while (wsi::getDisplayMode(wsi::getDefaultMonitor(), modeIndex++, &devMode)) {
       // Skip interlaced modes altogether
-      if (devMode.dmDisplayFlags & DM_INTERLACED)
+      if (devMode.interlaced)
         continue;
 
       // Skip modes with incompatible formats
-      if (devMode.dmBitsPerPel != GetMonitorFormatBpp(Format))
+      if (devMode.bitsPerPixel != GetMonitorFormatBpp(Format))
         continue;
 
-      if (!forcedRatio.undefined() && Ratio<DWORD>(devMode.dmPelsWidth, devMode.dmPelsHeight) != forcedRatio)
+      if (!forcedRatio.undefined() && Ratio<DWORD>(devMode.width, devMode.height) != forcedRatio)
         continue;
 
       D3DDISPLAYMODEEX mode;
       mode.Size             = sizeof(D3DDISPLAYMODEEX);
-      mode.Width            = devMode.dmPelsWidth;
-      mode.Height           = devMode.dmPelsHeight;
-      mode.RefreshRate      = devMode.dmDisplayFrequency;
+      mode.Width            = devMode.width;
+      mode.Height           = devMode.height;
+      mode.RefreshRate      = devMode.refreshRate.numerator / devMode.refreshRate.denominator;
```

Comment:
again why not use `ConvertDisplayMode` helper?

Might as well respect interlacing while we're at it (`D3DSCANLINEORDERING_INTERLACED`)